### PR TITLE
bf: S3C-3303 allow empty notif config

### PR DESCRIPTION
**Pull request by dora-korpar**
Fix to allow empty bucket notification to be put to bucket, effectively to delete existing configurations.

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/bucket/putBucketNotification.js [hunk 1]
```diff
@@ -89,6 +89,17 @@ describe('aws-sdk test put notification configuration', () => {
...
 
+        it('should put empty notification configuration', done => {
+            const params = {
+                Bucket: bucket,
+                NotificationConfiguration: {},
+            };
+            s3.putBucketNotificationConfiguration(params, err => {
+                assert.ifError(err);
+                done();
+            });
```
Comment (miniscruff) [inline tests/functional/aws-node-sdk/test/bucket/putBucketNotification.js]: Mocha will handle the error in the done callback so you can just pass that straight in.
```suggestion
            s3.putBucketNotificationConfiguration(params, done);
```

**Review by naren-rajendran — APPROVED** [review]

File: tests/unit/api/bucketPutNotification.js [hunk 2]
```diff
@@ -61,4 +66,16 @@ describe('putBucketNotification API', () => {
...
     });
+
+    it('should update bucket metadata with empty bucket notification', done => {
+        bucketPutNotification(authInfo, getNotifRequest(true), log, err => {
+            assert.ifError(err);
+            return metadata.getBucket(bucketName, log, (err, bucket) => {
+                assert.ifError(err);
+                const bucketNotifConfig = bucket.getNotificationConfiguration();
+                assert.deepStrictEqual(bucketNotifConfig, {});
+                return done();
```
Comment (miniscruff) [inline tests/unit/api/bucketPutNotification.js]: No need to return done, just calling it works.
I had a big spiel about not needing the `assert.ifErrors`'s but will save that for later...

Comment (dora-korpar) [reply tests/unit/api/bucketPutNotification.js]: not sure how big of a difference it makes, but i removed the returns

**Review by miniscruff — APPROVED** [review]

**Comment by dora-korpar** [comment]
@bert-e reset

**Comment by dora-korpar** [comment]
@bert-e reset

**Comment by dora-korpar** [comment]
@bert-e approve

**Pull request merged**